mqtt-client/src/main/java/com/gojek/mqtt/model/MqttConnectOptions.kt

```diff
@@ -1,18 +1,216 @@
+            builder.keepAlive.timeSeconds + 60
+        } else {
+            builder.readTimeoutSecs
+        }
+    }
+
+    fun newBuilder(): Builder = Builder(this)
+
+    class Builder() {
+        internal var serverUris: List<ServerUri> = emptyList()
+        internal var keepAlive: KeepAlive = NO_KEEP_ALIVE
+        internal var clientId: String = ""
+        internal var username: String = ""
+        internal var password: String = ""
+        internal var isCleanSession: Boolean = false
+        internal var readTimeoutSecs: Int = DEFAULT_READ_TIMEOUT
+        internal var version: MqttVersion = VERSION_3_1_1
+        internal var userPropertiesMap: Map<String, String> = emptyMap()
+        internal var socketFactory: SocketFactory = SocketFactory.getDefault()
+        internal var sslSocketFactoryOrNull: SSLSocketFactory? = null
+        internal var x509TrustManagerOrNull: X509TrustManager? = null
+        internal var connectionSpec: ConnectionSpec = DEFAULT_CONNECTION_SPECS
+        internal var protocols: List<Protocol> = emptyList()
+
+        internal constructor(mqttConnectOptions: MqttConnectOptions) : this() {
+            this.serverUris = mqttConnectOptions.serverUris
+            this.keepAlive = mqttConnectOptions.keepAlive
+            this.clientId = mqttConnectOptions.clientId
+            this.username = mqttConnectOptions.username
+            this.password = mqttConnectOptions.password
+            this.isCleanSession = mqttConnectOptions.isCleanSession
+            this.readTimeoutSecs = mqttConnectOptions.readTimeoutSecs
+            this.version = mqttConnectOptions.version
+            this.userPropertiesMap = mqttConnectOptions.userPropertiesMap
+            this.socketFactory = mqttConnectOptions.socketFactory
+            this.sslSocketFactoryOrNull = mqttConnectOptions.sslSocketFactory
+            this.x509TrustManagerOrNull = mqttConnectOptions.x509TrustManager
+            this.connectionSpec = mqttConnectOptions.connectionSpec
+            this.protocols = mqttConnectOptions.protocols
+        }
+
+        fun serverUris(serverUris: List<ServerUri>) = apply {
+            this.serverUris = serverUris
+        }
+
+        fun keepAlive(keepAlive: KeepAlive) = apply {
+            this.keepAlive = keepAlive
+        }
+
+        fun clientId(clientId: String) = apply {
+            this.clientId = clientId
+        }
+
+        fun userName(username: String) = apply {
+            this.username = username
+        }
+
+        fun password(password: String) = apply {
+            this.password = password
+        }
+
+        fun cleanSession(cleanSession: Boolean) = apply {
+            this.isCleanSession = cleanSession
+        }
+
+        fun readTimeoutSecs(readTimeoutSecs: Int) = apply {
+            require(readTimeoutSecs > 0 || readTimeoutSecs == DEFAULT_READ_TIMEOUT) { "read timeout should be >= $DEFAULT_READ_TIMEOUT" }
```

Comment:
can we simply make it `require(readTimeoutSecs > 0)`?

Reply:
done